lib/pure/strutils.nim

```diff
@@ -1370,17 +1371,69 @@ when not defined(js):
   proc c_sprintf(buf, frmt: cstring): cint {.header: "<stdio.h>",
                                      importc: "sprintf", varargs, noSideEffect.}
 
+# Forward declaration due to formatEng's use of ffDecimal formatting
+proc formatEng*(f: BiggestFloat,
+                precision: range[0..32] = 10,
+                decimalSep = '.',
+                trim: bool = true,
```

Comment:
No, it is just as weird to me. ``0.00`` means "precise up to 2 decimal places after the comma", the float vs int distinction wasn't my concern.